graylog2-server/src/main/java/org/graylog/events/notifications/NotificationResourceHandler.java

```diff
@@ -156,6 +171,93 @@ public boolean delete(String dtoId) {
         return notificationService.delete(dtoId) > 0;
     }
 
+    /**
+     * Tests a notification definition by executing it with a dummy event.
+     *
+     * @param notificationDto the notification definition to test
+     * @param userName the name of the user that triggered the test
+     * @throws NotFoundException if the notification definition or the notification factory cannot be found
+     */
+    public void test(NotificationDto notificationDto, String userName) throws NotFoundException {
+        final EventNotification.Factory eventNotificationFactory = eventNotificationFactories.get(notificationDto.config().type());
+        if (eventNotificationFactory == null) {
+            throw new NotFoundException("Couldn't find factory for notification type <" + notificationDto.config().type() + ">");
+        }
+        final EventNotificationContext notificationContext = getDummyContext(notificationDto, userName);
+        final EventNotification eventNotification = eventNotificationFactory.create();
+        try {
+            eventNotification.execute(notificationContext);
+        } catch (EventNotificationException e) {
+            throw new InternalServerErrorException(e.getMessage(), e);
+        }
+    }
+
+    private EventNotificationContext getDummyContext(NotificationDto notificationDto, String userName) {
+        final EventDto eventDto = EventDto.builder()
+                .alert(true)
+                .eventDefinitionId("EventDefinitionTestId")
```

Comment:
All the in-line literals look a bit rough. What about encapsulate them in a TestData class?